### PR TITLE
Buffer counters

### PR DESCRIPTION
**Pull request by AshokDaparthi**

### CODE REVIEW

**Comment by lguohan** [comment]
@itaibaz @tushar-ty @atitjain  please review

File: inc/saibuffer.h [hunk 1]
```diff
@@ -90,6 +90,9 @@ typedef enum _sai_ingress_priority_group_stat_t
     /** get watermark pg xoff room occupancy in bytes [uint64_t] */
     SAI_INGRESS_PRIORITY_GROUP_STAT_XOFF_ROOM_WATERMARK_BYTES = 0x00000007,
 
+    /** get/set rx bytes count [uint64_t] */
```
Comment (lguohan) [inline inc/saibuffer.h]: only get?

File: inc/saibuffer.h [hunk 2]
```diff
@@ -242,6 +245,12 @@ typedef enum _sai_buffer_pool_stat_t
     /** get watermark pool occupancy in bytes [uint64_t] */
     SAI_BUFFER_POOL_STAT_WATERMARK_BYTES = 0x00000001,
 
+    /** get count of bytes passed pool [uint64_t] */
```
Comment (lguohan) [inline inc/saibuffer.h]: what does bytes passed pool mean here?

File: inc/saibuffer.h [hunk 3]
```diff
@@ -242,6 +245,12 @@ typedef enum _sai_buffer_pool_stat_t
...
+    SAI_BUFFER_POOL_STAT_BYTES  = 0x00000002,
+
+    /** get count of bytes dropped by pool limits [uint64_t] */
```
Comment (lguohan) [inline inc/saibuffer.h]: only due to pool limits?

**Comment by lguohan** [comment]
@itaibaz, @atitjain , @tushar-ty , @krambn, please review?

**Comment by AshokDaparthi** [comment]
@lguohan - Please review

File: inc/saibuffer.h [hunk 4]
```diff
@@ -316,6 +325,20 @@ typedef sai_status_t(*sai_get_buffer_pool_stats_fn)(
...
+ * @param[in] pool_id Buffer pool id
+ * @param[in] counter_ids Specifies the array of counter ids
+ * @param[in] number_of_counters Number of counters in the array
+ *
+ * @return #SAI_STATUS_SUCCESS on success Failure status code on error
+ */
+typedef sai_status_t(*sai_clear_buffer_pool_stats_fn)(
+        _In_ sai_object_id_t pool_id,
+        _In_ const sai_buffer_pool_stat_t *counter_ids,
+        _In_ uint32_t number_of_counters);
```
Comment (kcudnik) [inline inc/saibuffer.h]: move count as second parrameter

**Comment by AshokDaparthi** [comment]
@kcudnik - done

File: inc/saibuffer.h [hunk 5]
```diff
@@ -251,6 +254,12 @@ typedef enum _sai_buffer_pool_stat_t
     /** get watermark pool occupancy in bytes [uint64_t] */
     SAI_BUFFER_POOL_STAT_WATERMARK_BYTES = 0x00000001,
 
+    /** get count of bytes uses pool [uint64_t] */
+    SAI_BUFFER_POOL_STAT_BYTES  = 0x00000002,
```
Comment (lguohan) [inline inc/saibuffer.h]: what's the difference bewteen ther CURR_OCCUPANCY_BYTES and this one?

File: inc/saiport.h [hunk 6]
```diff
@@ -1360,17 +1360,35 @@ typedef enum _sai_port_stat_t
...
+    SAI_PORT_STAT_OUT_WATERMARK_BYTES,
+
+    /** get out port current shared occupancy in bytes [uint64_t] */
+    SAI_PORT_STAT_OUT_SHARED_CURR_OCCUPANCY_BYTES,
+
+    /** get out port watermark shared occupancy in bytes [uint64_t] */
+    SAI_PORT_STAT_OUT_SHARED_WATERMARK_BYTES,
+
+    /** get in port packet drops due to buffers [uint64_t] */
+    SAI_PORT_STAT_IN_DROPPED_BYTES,
```
Comment (lguohan) [inline inc/saiport.h]: packet drops usually counts as packets

File: inc/saibuffer.h [hunk 7]
```diff
@@ -251,6 +254,12 @@ typedef enum _sai_buffer_pool_stat_t
...
+
+    /** get count of bytes dropped in this pool [uint64_t] */
+    SAI_BUFFER_POOL_STAT_DROPPED_BYTES  = 0x00000003,
```
Comment (lguohan) [inline inc/saibuffer.h]: we usually count drops in packets not in bytes.

File: inc/saibuffer.h [hunk 8]
```diff
@@ -91,6 +91,9 @@ typedef enum _sai_ingress_priority_group_stat_t
     /** get watermark pg xoff room occupancy in bytes [uint64_t] */
     SAI_INGRESS_PRIORITY_GROUP_STAT_XOFF_ROOM_WATERMARK_BYTES = 0x00000007,
 
+    /** get dropped bytes count [uint64_t] */
+    SAI_INGRESS_PRIORITY_GROUP_STAT_DROPPED_BYTES = 0x00000008,
```
Comment (lguohan) [inline inc/saibuffer.h]: -> packets

**Comment by lguohan** [comment]
@itaibaz , can you take a look at the dropped counters, we usually count those dropped as packets.

File: inc/saibuffer.h (hunk 8 above)
Comment (AshokDaparthi) [inline inc/saibuffer.h]: modified to packets

File: inc/saiport.h (hunk 6 above)
Comment (AshokDaparthi) [inline inc/saiport.h]: modified to packets

**Pull request merged**